core/src/main/scala/io/projectglow/bgen/BgenGenotypeReader.scala

```diff
@@ -0,0 +1,55 @@
+package io.projectglow.bgen
+
+import java.io.DataInput
+import java.util.zip.Inflater
+
+import com.github.luben.zstd.{Zstd => ZstdCodec}
+
+trait BgenGenotypeReader {
+  /**
+   * Read genotypes from a bgen [[DataInput]] into a byte array. Returns the uncompressed genotype bytes.
+   * Note that the genotypes returned are still encoded using the standard BGEN encoding scheme.
+   */
+  def readGenotypes(dataStream: DataInput): Array[Byte]
```

Comment:
Can we rename this to `readGenotypeBlock`? We should also be clear this is only for Layout 2 blocks.